src/getAnimalMap.js

```diff
@@ -1,7 +1,25 @@
 const data = require('../data/zoo_data');
 
-function getAnimalMap(options) {
-  // seu código aqui
+function createAnimalList(options, specie) {
+  return specie.residents.reduce((arr, animal) => {
+    if (options.sex && animal.sex === options.sex) arr.push(animal.name);
+    if (!options.sex) arr.push(animal.name);
+    return arr;
+  }, []);
```

Comment:
Um `reduce` semanticamente busca reduzir um `array` a um valor. Nós conseguimos criar um array diferente, como você fez, mas construi-lo com um `reduce` não carrega muito significado semântico. 

Se a gente for analisar seu código, você:

1. Pega alguns `residents` de acordo com as `options`
2. Mais especificamente, pega o `name` desse `resident`

Pensando nisso, sugiro o seguinte:
```suggestion
  return specie.residents
    .filter((resident) => (options.sex && resident.sex === options.sex) || (!options.sex))
    .map((resident) => resident.name)
```

Pensando novamente no que é feito:
1. Filtramos os residentes, nos baseando nas `options`
2. Mapeamos os residentes, pegando só os seus nomes

Reply:
Realmente fica muito melhor assim, não pensei em trocar os if's por um filter mas faz muito sentido. Eu estava ficando tão acostumada com reduce que queria usar para tudo!
Muito obrigada pelo feedback!